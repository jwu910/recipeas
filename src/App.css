html, body {
  color: #333;
}

.recipe-modal-content {
  border: 1px solid #f2f2f2;
  background-color: #fff;
  padding: 20px;
}

.myModal-header {
  border-bottom: 1px solid #c3c3c3;
  padding-bottom: 10px;
}

.recipe-title:hover {
  cursor: pointer;
  color: #55dfb6;
}
