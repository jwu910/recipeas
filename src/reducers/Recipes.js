import { ADD_RECIPE, EDIT_RECIPE, REMOVE_RECIPE } from '../actions';

export const defaultRecipes = [
  {
    title: 'Pumpkin Pie',
    ingredients: 'Pumpkin Puree, Sweetened Condensed Milk, Eggs, Pumpkin Pie Spice, Pie Crust'
  },
  {
    title: 'Spaghetti',
    ingredients: 'Noodles, Pasta Sauce, Meatballs'
  },
  {
    title: 'Peanut Butter Mug Cake',
    ingredients: 'Peanut Butter, Eggs, Baking Powder, Sugar'
  }
];

const Recipes = (state = defaultRecipes, action) => {
  switch (action.type) {
    case ADD_RECIPE:
      return [
        ...state,
        action.recipe
      ]
    case EDIT_RECIPE:
      return state.map(recipe => {
        return recipe.title === action.targetTitle ?
          action.recipe
          : recipe;
      });
    case REMOVE_RECIPE:
      return state.filter(recipe => recipe.title !== action.recipeTitle);
    default:
      return state;
  }
}

export default Recipes;
