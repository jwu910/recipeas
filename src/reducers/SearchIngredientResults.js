import {
  ADD_SEARCH_INGREDIENT_RESULTS,
  CLEAR_SEARCH_INGREDIENT_RESULTS
} from '../actions/actionTypes';

const searchIngredientResults = (state = {}, action) => {
  switch (action.type) {
    case ADD_SEARCH_INGREDIENT_RESULTS:
      return Object.assign({}, state, action.results);
    case CLEAR_SEARCH_INGREDIENT_RESULTS:
      return {};
    default:
      return state;
  }
};

export default searchIngredientResults;
