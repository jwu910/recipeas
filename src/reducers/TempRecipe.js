import {
  MODIFY_TEMP_TITLE,
  ADD_TEMP_INGREDIENT,
  EDIT_TEMP_INGREDIENT,
  REMOVE_TEMP_INGREDIENT,
  CLEAR_TEMP_RECIPE
} from '../actions';

const initialTempRecipe = {
  titleDetails: {
    title: '',
    allergens: '',
    servings: ''
  },
  ingredients: []
};

const TempRecipe = (state = initialTempRecipe, action) => {
  switch(action.type) {
    // works for adding or editing
    case MODIFY_TEMP_TITLE:
      return Object.assign({}, state, action.titleDetails);
    case ADD_TEMP_INGREDIENT:
      return Object.assign({}, state, {
        ingredients: [
          ...state.ingredients,
          action.ingredient
        ]
      });
    case EDIT_TEMP_INGREDIENT:
      return Object.assign({}, state, {
        ingredients: state.ingredients.map(ingredient => {
          return ingredient.id === action.id ?
            { id: ingredient.id, ...action.ingredient }
            : ingredient;
        })
      });
    case REMOVE_TEMP_INGREDIENT:
      return Object.assign({}, state, {
        ingredients: state.ingredients.filter(ingredient => ingredient.id !== action.id)
      });
    case CLEAR_TEMP_RECIPE:
      return {};
    default:
      return state;
  }
};

export default TempRecipe;
