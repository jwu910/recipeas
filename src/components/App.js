import React from 'react';
import Navigation from './Nav';
import Main from './Main';
import 'bootstrap/dist/css/bootstrap.css';
import '../App.css';

const App = (props) => {
  return (
    <div className="App">
      <Navigation />
      <Main />
    </div>
  );
};

export default App;
