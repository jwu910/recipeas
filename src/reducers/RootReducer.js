import { combineReducers } from 'redux';
import Recipes from './Recipes';
import TempRecipe from './TempRecipe';
import SearchIngredientResults from './SearchIngredientResults';

const RootReducer = combineReducers({
  recipes: Recipes,
  tempRecipe: TempRecipe,
  results: SearchIngredientResults
});

export default RootReducer;
