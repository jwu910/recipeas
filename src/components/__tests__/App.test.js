import React from 'react';
import App from '../App';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { defaultRecipes } from '../../reducers/Recipes';
import { MemoryRouter } from 'react-router-dom';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';

const store = configureStore()({
  recipes: defaultRecipes
});

describe ('App', () => {
  test('renders without crashing', () => {
    const wrapper = shallow(
      <Provider store={store}>
        <MemoryRouter>
          <App />
        </MemoryRouter>
      </Provider>
    );
    const tree = toJson(wrapper);

    expect(tree).toMatchSnapshot();
  });
});
