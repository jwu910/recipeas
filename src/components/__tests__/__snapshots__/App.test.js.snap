// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing 1`] = `
<div
  className="App"
>
  <nav
    className="navbar navbar-default navbar-fixed-top"
  >
    <div
      className="container-fluid"
    >
      <a
        className="navbar-brand"
        href="/"
      >
        Recipeas
      </a>
      <ul
        className="nav navbar-nav"
      >
        <li>
          <a
            href="https://github.com/phuchle/recipeas"
          >
            Source Code
          </a>
        </li>
      </ul>
    </div>
  </nav>
  <div
    className="container"
  >
    <h1>
      Recipes
    </h1>
    <ul
      className="list-group"
    >
      <li
        className="list-group-item"
      >
        <div
          className="recipe-container"
        >
          <h4
            className="recipe-title"
            onClick={[Function]}
          >
            Pumpkin Pie
          </h4>
          <div
            aria-expanded={null}
            className="collapse"
          >
             
            <div
              className="well"
            >
              <ul>
                <span
                  className="list-group-item"
                >
                  Pumpkin Puree
                </span>
                <span
                  className="list-group-item"
                >
                   Sweetened Condensed Milk
                </span>
                <span
                  className="list-group-item"
                >
                   Eggs
                </span>
                <span
                  className="list-group-item"
                >
                   Pumpkin Pie Spice
                </span>
                <span
                  className="list-group-item"
                >
                   Pie Crust
                </span>
              </ul>
              <button
                className="edit-recipe btn btn-default"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                Edit
              </button>
              <button
                className="delete-recipe btn btn-danger"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                Delete
              </button>
            </div>
          </div>
        </div>
      </li>
      <li
        className="list-group-item"
      >
        <div
          className="recipe-container"
        >
          <h4
            className="recipe-title"
            onClick={[Function]}
          >
            Spaghetti
          </h4>
          <div
            aria-expanded={null}
            className="collapse"
          >
             
            <div
              className="well"
            >
              <ul>
                <span
                  className="list-group-item"
                >
                  Noodles
                </span>
                <span
                  className="list-group-item"
                >
                   Pasta Sauce
                </span>
                <span
                  className="list-group-item"
                >
                   Meatballs
                </span>
              </ul>
              <button
                className="edit-recipe btn btn-default"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                Edit
              </button>
              <button
                className="delete-recipe btn btn-danger"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                Delete
              </button>
            </div>
          </div>
        </div>
      </li>
      <li
        className="list-group-item"
      >
        <div
          className="recipe-container"
        >
          <h4
            className="recipe-title"
            onClick={[Function]}
          >
            Peanut Butter Mug Cake
          </h4>
          <div
            aria-expanded={null}
            className="collapse"
          >
             
            <div
              className="well"
            >
              <ul>
                <span
                  className="list-group-item"
                >
                  Peanut Butter
                </span>
                <span
                  className="list-group-item"
                >
                   Eggs
                </span>
                <span
                  className="list-group-item"
                >
                   Baking Powder
                </span>
                <span
                  className="list-group-item"
                >
                   Sugar
                </span>
              </ul>
              <button
                className="edit-recipe btn btn-default"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                Edit
              </button>
              <button
                className="delete-recipe btn btn-danger"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                Delete
              </button>
            </div>
          </div>
        </div>
      </li>
    </ul>
    <button
      className="btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Add Recipe
    </button>
  </div>
</div>
`;
